strace.py

```diff
@@ -167,10 +167,13 @@ def syscallTrace(self, process):
                 event = self.debugger.waitSyscall()
             except ProcessExit as event:
                 self.processExited(event)
+                if event.exitcode is not None:
+                    exitcode = event.exitcode
                 continue
             except ProcessSignal as event:
                 event.display()
                 event.process.syscall(event.signum)
+                exitcode = event.signum
```

Comment:
Should it be a negative number?

Reply:
Exit codes are in the range `[0-255]`, anyway I forgot to cast the `signal.XXX` objects. I straced an infinite loop in c, before killing it with a SIGSEGV, the exit code was `139`, as `127 + 11 (SIGSEGV)`, so I guess I should return `127 + int(event.signum)` ?

Reply:
"When a command terminates on a fatal signal whose number is N, Bash uses the value 128+N as the exit status." https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html

I suggest to add a function somewhere in the ptrace module which converts a signal number to an exit code. 128 + signum sounds like a good plan.

sys.exit(signum) sounds wrong to me. For example, SIGHUP is 1 on Linux. It would be easy to by confused between a regular exit with code 1 and a process killed by SIGHUP.